### PR TITLE
[bug] Fix GraphQL Query for Catalog content

### PR DESCRIPTION
**Pull request by theBeginner86**
**Notes for Reviewers**

This PR fixes the GQL Query for catalog content

https://github.com/meshery/meshery/assets/73700530/b2c3e35a-0e67-460c-8701-a88240d5511c

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/Filters.js [hunk 1]
```diff
@@ -476,6 +478,7 @@ function MesheryFilters({ updateProgress, enqueueSnackbar, closeSnackbar, user,
       disposeConfSubscriptionRef.current.dispose();
     }
     const configurationSubscription = ConfigurationSubscription((result) => {
+      console.log("FilterFile subscription: ", result)
```
Comment (theBeginner86) [inline ui/components/Filters.js:481]:
```suggestion
     
```

**Review by leecalcote — APPROVED** [review]
Whoa. `pagesize` got us here. A good note to get in front of @abdullah1308

**Comment by leecalcote** [comment]
Golang build check failures are unrelated to this PR. @Abhishek-kumar09 will need to address separately.

**Comment by leecalcote** [comment]
Here's what I get when I build and open the Performance page:

```
INFO[0062] attempting to fetch results from cloud
DEBU[0062] constructed results url: https://meshery.layer5.io/api/performance/results?page=0&pagesize=10
ERRO[0062] error while fetching results: invalid performance profile id
runtime error: invalid memory address or nil pointer dereference

goroutine 4832 [running]:
runtime/debug.Stack()
	/opt/homebrew/Cellar/go/1.19.6/libexec/src/runtime/debug/stack.go:24 +0x64
runtime/debug.PrintStack()
	/opt/homebrew/Cellar/go/1.19.6/libexec/src/runtime/debug/stack.go:16 +0x1c
github.com/99designs/gqlgen/graphql.DefaultRecover({0x14000e9b5c0?, 0x0?}, {0x105461ae0?, 0x10757d290?})
	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/recovery.go:17 +0x78
github.com/99designs/gqlgen/graphql.(*OperationContext).Recover(0x0?, {0x1059c1c30, 0x1400080cc00}, {0x105461ae0?, 0x10757d290?})
	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/context_operation.go:124 +0x44
github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query_fetchAllResults.func1()
	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:13049 +0x68
panic({0x105461ae0, 0x10757d290})
	/opt/homebrew/Cellar/go/1.19.6/libexec/src/runtime/panic.go:884 +0x204
github.com/layer5io/meshery/server/models.ErrFetch({0x0, 0x0}, {0x14000aea080, 0x76}, 0x14000e6e378?)
	/Users/lee/code/meshery/server/models/error.go:173 +0x60
github.com/layer5io/meshery/server/models.(*RemoteProvider).FetchAllResults(0x14000df41c0, {0x14001edd70d, 0x717}, {0x10741c480, 0x1}, {0x1400244ad20, 0x2}, {0x0, 0x0}, {0x0, ...}, ...)
	/Users/lee/code/meshery/server/models/remote_provider.go:953 +0xaa8
github.com/layer5io/meshery/server/internal/graphql/resolver.(*Resolver).fetchAllResults(0x14000aa8cc0, {0x1059c1c30?, 0x1400080cc00?}, {0x1059f0f08, 0x14000df41c0}, {{0x10741c480, 0x1}, {0x1400244ad20, 0x2}, 0x1400092a3d0, ...})
	/Users/lee/code/meshery/server/internal/graphql/resolver/performance.go:191 +0xdc
github.com/layer5io/meshery/server/internal/graphql/resolver.(*queryResolver).FetchAllResults(0x140036f4cc8, {0x1059c1c30, 0x1400080cc00}, {{0x10741c480, 0x1}, {0x1400244ad20, 0x2}, 0x1400092a3d0, 0x1400092a450, 0x1400092a4d0, ...})
	/Users/lee/code/meshery/server/internal/graphql/resolver/schema.resolvers.go:126 +0xc8
github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query_fetchAllResults.func2({0x1059c1c30?, 0x1400080cc00})
	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:13055 +0x124
github.com/99designs/gqlgen/graphql/executor.processExtensions.func4({0x1059c1c30?, 0x1400080cc00?}, 0x1400020ad90?)
	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/executor/extensions.go:72 +0x2c
github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query_fetchAllResults(0x14000210850, {0x1059c1c30, 0x1400080c870}, {0x14000546a00?, {0x14000b0bc80?, 0x10248d3e8?, 0x14000e6e748?}})
	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:13053 +0x1dc
github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query.func40({0x1059c1c30?, 0x1400080c870?})
	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:19348 +0x8c
github.com/99designs/gqlgen/graphql/executor.processExtensions.func3({0x1059c1c30?, 0x1400080c870?}, 0x108a4cf18?)
	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/executor/extensions.go:69 +0x2c
github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query.func41({0x1059c1c30?, 0x1400080c870?})
	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:19356 +0x38
github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query.func42()
	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:19360 +0x2c
github.com/99designs/gqlgen/graphql.(*FieldSet).Dispatch(0x140007760f0)
	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/fieldset.go:34 +0x94
github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query(0x14000210850, {0x1059c1c30, 0x1400080c7e0}, {0x14000cf0ad0, 0x1, 0x1})
	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:19539 +0x1f4
github.com/layer5io/meshery/server/internal/graphql/generated.(*executableSchema).Exec.func1({0x1059c1c30?, 0x1400080c7b0?})
	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:2277 +0x8c
github.com/99designs/gqlgen/graphql/executor.(*Executor).DispatchOperation.func1.1.1({0x1059c1c30, 0x1400080c7b0})
	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/executor/executor.go:119 +0x2c
github.com/99designs/gqlgen/graphql/executor.processExtensions.func2({0x1059c1c30?, 0x1400080c7b0?}, 0x10527bf00?)
	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/executor/extensions.go:66 +0x2c
github.com/99designs/gqlgen/graphql/executor.(*Executor).DispatchOperation.func1.1({0x1059c1c30, 0x1400080c6f0})
	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/executor/executor.go:118 +0x118
github.com/99designs/gqlgen/graphql/handler/transport.POST.Do({0x1059c1c30?}, {0x1059bf088?, 0x140037f5340}, 0x14004619000, {0x1059ad238, 0x1400013a1e0})
	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/handler/transport/http_post.go:91 +0x420
github.com/99designs/gqlgen/graphql/handler.(*Server).ServeHTTP(0x14000372840, {0x1059bf088, 0x140037f5340}, 0x14004619000)
	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/handler/server.go:121 +0x194
github.com/layer5io/meshery/server/handlers.(*Handler).GraphqlMiddleware.func1({0x1059bf088?, 0x140037f5340?}, 0x10527d500?, 0x10597c3c0?, 0x105422200?, {0x14000bf0f90?, 0x108a4ca68?})
	/Users/lee/code/meshery/server/handlers/middlewares.go:265 +0x40
github.com/layer5io/meshery/server/handlers.(*Handler).MesheryControllersMiddleware.func1({0x1059bf088, 0x140037f5340}, 0x140023d0700, 0x10597c3a0?, 0x1051f79e0?, {0x1059f0f08, 0x14000df41c0})
	/Users/lee/code/meshery/server/handlers/middlewares.go:138 +0x230
github.com/layer5io/meshery/server/handlers.(*Handler).KubernetesMiddleware.func1({0x1059bf088, 0x140037f5340}, 0x14000395600, 0x10597c440?, 0x1056306c0?, {0x1059f0f08, 0x14000df41c0?})
	/Users/lee/code/meshery/server/handlers/middlewares.go:205 +0x8e4
github.com/layer5io/meshery/server/handlers.(*Handler).SessionInjectorMiddleware.func1({0x1059bf088, 0x140037f5340}, 0x14000361b00)
	/Users/lee/code/meshery/server/handlers/middlewares.go:258 +0x4d4
net/http.HandlerFunc.ServeHTTP(0x1400200aab0?, {0x1059bf088?, 0x140037f5340?}, 0x100?)
	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:2109 +0x38
github.com/layer5io/meshery/server/handlers.(*Handler).AuthMiddleware.func1({0x1059bf088?, 0x140037f5340?}, 0x14000361b00?)
	/Users/lee/code/meshery/server/handlers/middlewares.go:102 +0x234
net/http.HandlerFunc.ServeHTTP(0x1059c1c30?, {0x1059bf088?, 0x140037f5340?}, 0x10597c390?)
	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:2109 +0x38
github.com/layer5io/meshery/server/handlers.(*Handler).ProviderMiddleware.func1({0x1059bf088, 0x140037f5340}, 0x14000361a00)
	/Users/lee/code/meshery/server/handlers/middlewares.go:54 +0x3d4
net/http.HandlerFunc.ServeHTTP(0x14000361900?, {0x1059bf088?, 0x140037f5340?}, 0x800?)
	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:2109 +0x38
github.com/gorilla/mux.(*Router).ServeHTTP(0x14000b88cc0, {0x1059bf088, 0x140037f5340}, 0x14000361800)
	/Users/lee/code/go/bin/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x19c
net/http.serverHandler.ServeHTTP({0x1059aa808?}, {0x1059bf088, 0x140037f5340}, 0x14000361800)
	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:2947 +0x2c4
net/http.(*conn).serve(0x14000f49400, {0x1059c1c30, 0x14000d621e0})
	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:1991 +0x560
created by net/http.(*Server).Serve
	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:3102 +0x444
INFO[0062] attempting to fetch performance profiles from cloud
```

**Comment by theBeginner86** [comment]
> Here's what I get when I build and open the Performance page:
> 
> ```
> INFO[0062] attempting to fetch results from cloud
> DEBU[0062] constructed results url: https://meshery.layer5.io/api/performance/results?page=0&pagesize=10
> ERRO[0062] error while fetching results: invalid performance profile id
> runtime error: invalid memory address or nil pointer dereference
> 
> goroutine 4832 [running]:
> runtime/debug.Stack()
> 	/opt/homebrew/Cellar/go/1.19.6/libexec/src/runtime/debug/stack.go:24 +0x64
> runtime/debug.PrintStack()
> 	/opt/homebrew/Cellar/go/1.19.6/libexec/src/runtime/debug/stack.go:16 +0x1c
> github.com/99designs/gqlgen/graphql.DefaultRecover({0x14000e9b5c0?, 0x0?}, {0x105461ae0?, 0x10757d290?})
> 	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/recovery.go:17 +0x78
> github.com/99designs/gqlgen/graphql.(*OperationContext).Recover(0x0?, {0x1059c1c30, 0x1400080cc00}, {0x105461ae0?, 0x10757d290?})
> 	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/context_operation.go:124 +0x44
> github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query_fetchAllResults.func1()
> 	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:13049 +0x68
> panic({0x105461ae0, 0x10757d290})
> 	/opt/homebrew/Cellar/go/1.19.6/libexec/src/runtime/panic.go:884 +0x204
> github.com/layer5io/meshery/server/models.ErrFetch({0x0, 0x0}, {0x14000aea080, 0x76}, 0x14000e6e378?)
> 	/Users/lee/code/meshery/server/models/error.go:173 +0x60
> github.com/layer5io/meshery/server/models.(*RemoteProvider).FetchAllResults(0x14000df41c0, {0x14001edd70d, 0x717}, {0x10741c480, 0x1}, {0x1400244ad20, 0x2}, {0x0, 0x0}, {0x0, ...}, ...)
> 	/Users/lee/code/meshery/server/models/remote_provider.go:953 +0xaa8
> github.com/layer5io/meshery/server/internal/graphql/resolver.(*Resolver).fetchAllResults(0x14000aa8cc0, {0x1059c1c30?, 0x1400080cc00?}, {0x1059f0f08, 0x14000df41c0}, {{0x10741c480, 0x1}, {0x1400244ad20, 0x2}, 0x1400092a3d0, ...})
> 	/Users/lee/code/meshery/server/internal/graphql/resolver/performance.go:191 +0xdc
> github.com/layer5io/meshery/server/internal/graphql/resolver.(*queryResolver).FetchAllResults(0x140036f4cc8, {0x1059c1c30, 0x1400080cc00}, {{0x10741c480, 0x1}, {0x1400244ad20, 0x2}, 0x1400092a3d0, 0x1400092a450, 0x1400092a4d0, ...})
> 	/Users/lee/code/meshery/server/internal/graphql/resolver/schema.resolvers.go:126 +0xc8
> github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query_fetchAllResults.func2({0x1059c1c30?, 0x1400080cc00})
> 	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:13055 +0x124
> github.com/99designs/gqlgen/graphql/executor.processExtensions.func4({0x1059c1c30?, 0x1400080cc00?}, 0x1400020ad90?)
> 	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/executor/extensions.go:72 +0x2c
> github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query_fetchAllResults(0x14000210850, {0x1059c1c30, 0x1400080c870}, {0x14000546a00?, {0x14000b0bc80?, 0x10248d3e8?, 0x14000e6e748?}})
> 	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:13053 +0x1dc
> github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query.func40({0x1059c1c30?, 0x1400080c870?})
> 	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:19348 +0x8c
> github.com/99designs/gqlgen/graphql/executor.processExtensions.func3({0x1059c1c30?, 0x1400080c870?}, 0x108a4cf18?)
> 	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/executor/extensions.go:69 +0x2c
> github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query.func41({0x1059c1c30?, 0x1400080c870?})
> 	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:19356 +0x38
> github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query.func42()
> 	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:19360 +0x2c
> github.com/99designs/gqlgen/graphql.(*FieldSet).Dispatch(0x140007760f0)
> 	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/fieldset.go:34 +0x94
> github.com/layer5io/meshery/server/internal/graphql/generated.(*executionContext)._Query(0x14000210850, {0x1059c1c30, 0x1400080c7e0}, {0x14000cf0ad0, 0x1, 0x1})
> 	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:19539 +0x1f4
> github.com/layer5io/meshery/server/internal/graphql/generated.(*executableSchema).Exec.func1({0x1059c1c30?, 0x1400080c7b0?})
> 	/Users/lee/code/meshery/server/internal/graphql/generated/generated.go:2277 +0x8c
> github.com/99designs/gqlgen/graphql/executor.(*Executor).DispatchOperation.func1.1.1({0x1059c1c30, 0x1400080c7b0})
> 	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/executor/executor.go:119 +0x2c
> github.com/99designs/gqlgen/graphql/executor.processExtensions.func2({0x1059c1c30?, 0x1400080c7b0?}, 0x10527bf00?)
> 	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/executor/extensions.go:66 +0x2c
> github.com/99designs/gqlgen/graphql/executor.(*Executor).DispatchOperation.func1.1({0x1059c1c30, 0x1400080c6f0})
> 	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/executor/executor.go:118 +0x118
> github.com/99designs/gqlgen/graphql/handler/transport.POST.Do({0x1059c1c30?}, {0x1059bf088?, 0x140037f5340}, 0x14004619000, {0x1059ad238, 0x1400013a1e0})
> 	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/handler/transport/http_post.go:91 +0x420
> github.com/99designs/gqlgen/graphql/handler.(*Server).ServeHTTP(0x14000372840, {0x1059bf088, 0x140037f5340}, 0x14004619000)
> 	/Users/lee/code/go/bin/pkg/mod/github.com/99designs/gqlgen@v0.17.28/graphql/handler/server.go:121 +0x194
> github.com/layer5io/meshery/server/handlers.(*Handler).GraphqlMiddleware.func1({0x1059bf088?, 0x140037f5340?}, 0x10527d500?, 0x10597c3c0?, 0x105422200?, {0x14000bf0f90?, 0x108a4ca68?})
> 	/Users/lee/code/meshery/server/handlers/middlewares.go:265 +0x40
> github.com/layer5io/meshery/server/handlers.(*Handler).MesheryControllersMiddleware.func1({0x1059bf088, 0x140037f5340}, 0x140023d0700, 0x10597c3a0?, 0x1051f79e0?, {0x1059f0f08, 0x14000df41c0})
> 	/Users/lee/code/meshery/server/handlers/middlewares.go:138 +0x230
> github.com/layer5io/meshery/server/handlers.(*Handler).KubernetesMiddleware.func1({0x1059bf088, 0x140037f5340}, 0x14000395600, 0x10597c440?, 0x1056306c0?, {0x1059f0f08, 0x14000df41c0?})
> 	/Users/lee/code/meshery/server/handlers/middlewares.go:205 +0x8e4
> github.com/layer5io/meshery/server/handlers.(*Handler).SessionInjectorMiddleware.func1({0x1059bf088, 0x140037f5340}, 0x14000361b00)
> 	/Users/lee/code/meshery/server/handlers/middlewares.go:258 +0x4d4
> net/http.HandlerFunc.ServeHTTP(0x1400200aab0?, {0x1059bf088?, 0x140037f5340?}, 0x100?)
> 	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:2109 +0x38
> github.com/layer5io/meshery/server/handlers.(*Handler).AuthMiddleware.func1({0x1059bf088?, 0x140037f5340?}, 0x14000361b00?)
> 	/Users/lee/code/meshery/server/handlers/middlewares.go:102 +0x234
> net/http.HandlerFunc.ServeHTTP(0x1059c1c30?, {0x1059bf088?, 0x140037f5340?}, 0x10597c390?)
> 	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:2109 +0x38
> github.com/layer5io/meshery/server/handlers.(*Handler).ProviderMiddleware.func1({0x1059bf088, 0x140037f5340}, 0x14000361a00)
> 	/Users/lee/code/meshery/server/handlers/middlewares.go:54 +0x3d4
> net/http.HandlerFunc.ServeHTTP(0x14000361900?, {0x1059bf088?, 0x140037f5340?}, 0x800?)
> 	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:2109 +0x38
> github.com/gorilla/mux.(*Router).ServeHTTP(0x14000b88cc0, {0x1059bf088, 0x140037f5340}, 0x14000361800)
> 	/Users/lee/code/go/bin/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x19c
> net/http.serverHandler.ServeHTTP({0x1059aa808?}, {0x1059bf088, 0x140037f5340}, 0x14000361800)
> 	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:2947 +0x2c4
> net/http.(*conn).serve(0x14000f49400, {0x1059c1c30, 0x14000d621e0})
> 	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:1991 +0x560
> created by net/http.(*Server).Serve
> 	/opt/homebrew/Cellar/go/1.19.6/libexec/src/net/http/server.go:3102 +0x444
> INFO[0062] attempting to fetch performance profiles from cloud
> ```

Yes, this is what I was talking about in slack, that I was getting some nil ptr error and Meshery servers was crashing. I have seen this.

**Comment by leecalcote** [comment]
Ooooh, I see. Ok. Please note this commit - https://github.com/meshery/meshery/pull/7985/commits/e583dd0c7c5b19ba9a8bee599753adc49e621cbd. Feel free to revert this commit, if not needed, @theBeginner86

**Comment by theBeginner86** [comment]
> Ooooh, I see. Ok. Please note this commit - [e583dd0](https://github.com/meshery/meshery/commit/e583dd0c7c5b19ba9a8bee599753adc49e621cbd). Feel free to revert this commit, if not needed, @theBeginner86

No, Its fine. We can merge this PR as the only failing tests are server build and graphql docs build.
Server build failure is not related to this PR and graphql docs build failure is the same error that I get locally. We can track it in a new issue. Its related to recent update of required ruby version from v2.. to v3..

**Pull request merged**